domain/miningmanager/factory.go

```diff
@@ -2,20 +2,21 @@ package miningmanager
 
 import (
 	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/dagconfig"
 	"github.com/kaspanet/kaspad/domain/miningmanager/blocktemplatebuilder"
 	mempoolpkg "github.com/kaspanet/kaspad/domain/miningmanager/mempool"
 )
 
 // Factory instantiates new mining managers
 type Factory interface {
-	NewMiningManager(consensus externalapi.Consensus, blockMaxMass uint64, acceptNonStd bool) MiningManager
+	NewMiningManager(consensus externalapi.Consensus, blockMaxMass uint64, acceptNonStd bool, params *dagconfig.Params) MiningManager
 }
 
 type factory struct{}
 
 // NewMiningManager instantiate a new mining manager
-func (f *factory) NewMiningManager(consensus externalapi.Consensus, blockMaxMass uint64, acceptNonStd bool) MiningManager {
-	mempool := mempoolpkg.New(consensus, acceptNonStd)
+func (f *factory) NewMiningManager(consensus externalapi.Consensus, blockMaxMass uint64, acceptNonStd bool, params *dagconfig.Params) MiningManager {
+	mempool := mempoolpkg.New(consensus, acceptNonStd, params)
```

Comment:
It appears that the only reason why you're passing the entire `params` into the mempool is for MaxMassAcceptedByBlock.
You've already passed MaxMassAcceptedByBlock into NewMiningManager, (it's `blockMaxMass`) so why not just pass that into the mempool instead?